js/gameFunctionality.js

```diff
@@ -1,219 +0,0 @@
-// Starts the game and checks for game over
-function gameStatus() {
-  if (player.y < height / 2) {
-    gameStarted = true;
-  }
-
-  if (gameStarted) {
-    gameSpeed = 1.5;
-  }
-
-  if (gameOver) {
-    black = 0; // Using these for color-effect on both start and gameover screen, thus the reset
-    white = 255;
-    gameover.play();
-    bgmusic.stop();
-  }
-}
-
-function drawScore() {
-  push();
-  strokeWeight(1);
-  fill(200, 200, 0, colorChange / 2 + 50);
-  circle(42, 42, 40);
-  textAlign(CENTER);
-  textFont(font);
-  fill(0);
-  textSize(18);
-  text(`${score}`, 42, 39);
-  pop();
-}
-
-function drawVolumeSlider() {
-  slider = createSlider(0, 0.6, 0.3, 0.02);
-  slider.position(50, 40);
-  slider.style("width", "80px");
-}
-
-function startScreen() {
-  push();
-  textFont(font);
-  background(0);
-  fill(white);
-  circle(width / 2, height / 2, circleD);
-  fill(255);
-  textAlign(CENTER);
-
-  if (startScreenCounter > 10) {
-    // startScreenCounter starts when "orb" appeared
-    textSize(75);
-    fill(black, alphaCount * 2); // using alphaCount to have a fading in
-    text("orb", width / 2, height / 2 + 25);
-  }
-
-  if (startScreenCounter > 50) {
-    textSize(40);
-    fill(white);
-    text("Press 'orb' to play", width / 2, height / 2 + 300);
-  }
-  pop();
-  // Growing/shrinking cycle
-  if (circleD > 200) {
-    grow = false;
-  }
-  if (circleD < 181) {
-    grow = true;
-  }
-  // different tempo at smallest and biggest size
-  if (grow == true) {
-    if (circleD > 197) {
-      circleD += growAmount / 4;
-      startScreenCounter++;
-    } else if (circleD < 180) {
-      circleD += growAmount;
-    } else {
-      circleD += growAmount / 2;
-      if (startScreenCounter > 0) alphaCount += 2; // "orb" fades in, after some time
-    }
-  } else {
-    if (circleD < 183) {
-      circleD -= growAmount / 4;
-    } else {
-      circleD -= growAmount / 3;
-    }
-  }
-}
-
-function gameOverScreen() {
-  push();
-  textFont(font);
-  background(255);
-  fill(black);
-  noStroke();
-  circle(width / 2, height / 2, gameOverCircleD);
-
-  if (gameOverCircleD > 200) {
-    gameOverCircleD -= 10;
-  }
-
-  if (gameOverCircleD < 201) {
-    textSize(40);
-    fill(white);
-    textAlign(CENTER);
-    fill(white, GameOverAlphaCount);
-    text("Game", width / 2, height / 2 - 25);
-    text("Over", width / 2, height / 2 + 15);
-    fill(black);
-    text(`Your Score is ${score}`, width / 2, height / 2 + 150);
-
-    if (GameOverAlphaCount < -600) {
-      textSize(75);
-      fill(white);
-      text("orb", width / 2, height / 2 - 15);
-      if (gameOverToggle1) {
-        infobop.play();
-        gameOverToggle1 = false;
-      }
-    }
-
-    if (GameOverAlphaCount < -1600) {
-      if (gameOverToggle2) {
-        infobop.play();
-        gameOverToggle2 = false;
-      }
-      textSize(40);
-      fill(black);
-      text("Press 'orb' to play again", width / 2, height / 2 + 300);
```

Comment:
width / 2 can be precomputed once and then stored in a variable -> i.e. your CPU has to do a lot less divisions unnecessarily. (this is a form of "caching" the value)

Reply:
same for height / 2

Reply:
stuff like this might actually significantly increase performance. depending on how often you actually do them you can get rid of quite a lot of CPU cycles that way ^^ and draw happens quite a lot :P

Reply:
though, exact measures of performance would have to be done with proper benchmarks ofc i.e. before after performance profiling... probably here it's not too bad, but if you unnecessarily recompute something like fibonacci numbers or square roots or things like those, well then it can drastically improve speed.